csharp/src/Microsoft.ML.OnnxRuntime/NamedOnnxValue.cs

```diff
@@ -71,8 +71,15 @@ public IEnumerable<T> AsEnumerable<T>()
         /// </summary>
         /// <param name="onnxValue"></param>
         /// <param name="pinnedMemoryHandle"></param>
-        internal void ToNativeOnnxValue(out IntPtr onnxValue, out MemoryHandle pinnedMemoryHandle)
+        /// <param name="disposeOnnxValueAfterUse"></param>
+        internal virtual void ToNativeOnnxValue(out IntPtr onnxValue, 
```

Comment:
Can we just make this class IDisposable instead of returning disposeOnnxValueAfterUse? 

What happens in Dispose is an implementation detail of this class or a derived class. Users shouldn't need to care - they can just call Dispose and the implementation should do the right thing for the instance. Expecting a user to track which instances need disposing feels backwards to me and an unnecessary burden on the user of the class.

Reply:
That is along the lines of what I had in the original PR wherein the native OrtValue was a member and there was method to clean it up after use. But as Yulong pointed out, that makes the class "stateful" and we cannot reliably dispose the instance in session Run() after use as the instance may be being used on another thread (the reason for the CI failures in my original PR). Making this class IDisposable is hence not an option as we would have to have the end user dispose this instance when it is safe to do so and this breaks backwards compatibility (and makes it not different from DisposableNamedOnnxValue !). Moreover ToNativeOnnxValue() is an internal method and we don't expect this to be used by the end user.

Reply:
[`IDisposable`](https://docs.microsoft.com/en-us/dotnet/api/system.idisposable?view=netframework-4.8) is a special interface that a C# user **do need to care**. An object that implemented `IDisposable` is expected users to call its Dispose() method (typically via using the keyword `using`).